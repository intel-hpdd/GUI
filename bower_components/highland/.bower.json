{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}