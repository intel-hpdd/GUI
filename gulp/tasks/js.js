//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
