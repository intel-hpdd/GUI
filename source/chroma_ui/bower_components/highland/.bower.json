{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}