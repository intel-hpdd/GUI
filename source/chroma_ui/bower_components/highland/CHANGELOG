Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
