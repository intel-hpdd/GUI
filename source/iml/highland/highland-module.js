// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
