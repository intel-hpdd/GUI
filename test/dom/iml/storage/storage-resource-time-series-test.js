// @flow

import { render } from "inferno";
import highland from "highland";
import mockD3 from "d3";
import { flushD3Transitions } from "../../../test-utils.js";

describe("storage resource time series", () => {
  let mockSocketStream, mockBufferDataNewerThan, mockGetTimeParams, mockStream, StorageResourceTimeSeries, root;

  beforeEach(() => {
    mockStream = highland();
    mockSocketStream = jest.fn(() => mockStream);

    mockGetTimeParams = {
      getRequestDuration: jest.fn(() => () => x => ({
        ...x,
        qs: {
          ...x.qs,
          begin: "2017-08-08T20:30:00.000Z",
          end: "2017-08-08T20:40:00.000Z"
        }
      }))
    };

    mockBufferDataNewerThan = jest.fn(() => x => x);

    jest.mock("d3", () => mockD3);

    jest.mock("../../../../source/iml/charting/buffer-data-newer-than.js", () => mockBufferDataNewerThan);

    jest.mock("../../../../source/iml/charting/get-time-params.js", () => ({
      getTimeParams: mockGetTimeParams
    }));

    jest.mock("../../../../source/iml/socket/socket-stream.js", () => mockSocketStream);

    StorageResourceTimeSeries = require("../../../../source/iml/storage/storage-resource-time-series.js").default;

    root = document.createElement("div");

    const data = {
      type: "timeseries",
      title: "my first timeseries",
      series: [
        {
          label: "reads",
          name: "reads",
          type: "timeseries",
          data: null,
          unit_name: "reads"
        }
      ]
    };

    render(<StorageResourceTimeSeries resourceUri="/api/foo/bar/" chart={data} />, root);
    // $FlowFixMe: Mock for test
    Element.prototype.getTotalLength = () => 100;
    jest.useFakeTimers();
  });

  afterEach(() => {
    // $FlowFixMe: Mock for test
    delete Element.prototype.getTotalLength;
    jest.clearAllTimers();
    jest.useRealTimers();
  });

  it("should render when fetching data", () => {
    expect(root).toMatchSnapshot();
  });

  it("should call the socketStream", () => {
    expect(mockSocketStream).toHaveBeenCalledOnceWith(
      "/api/foo/bar/metric/",
      {
        qs: {
          begin: "2017-08-08T20:30:00.000Z",
          end: "2017-08-08T20:40:00.000Z",
          metrics: "reads"
        }
      },
      true
    );
  });

  it("should render empty data", () => {
    mockStream.write([]);
    mockStream.end();
    jest.runAllTimers();
    flushD3Transitions(mockD3);

    expect(root).toMatchSnapshot();
  });

  it("should render data", () => {
    mockStream.write([
      { data: { reads: 3 }, ts: "2017-08-08T20:32:40+00:00" },
      { data: { reads: 4 }, ts: "2017-08-08T20:32:50+00:00" }
    ]);
    mockStream.end();

    jest.runAllTimers();
    flushD3Transitions(mockD3);

    expect(root).toMatchSnapshot();
  });
});
