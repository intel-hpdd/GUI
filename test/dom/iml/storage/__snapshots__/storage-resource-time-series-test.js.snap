// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage resource time series should render data 1`] = `
<div>
  <div
    style="height: 500px; display: flex;"
    class="storage-detail-chart"
  >
    <div
      style="display: flex;"
    >
      <h5
        style="width: 24px;"
      >
        reads
      </h5>
    </div>
    <div>
      <svg
        style="width: 100%; height: 100%;"
        class="charting"
      >
        <g
          transform="translate(50,50)"
          class="charting-group"
        >
          <g
            class="axis x-axis"
            transform="translate(0,-100)"
          >
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :40
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-8,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :41
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-16,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :42
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-24,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :43
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-32,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :44
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-40,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :45
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-48,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :46
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-56,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :47
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-64,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :48
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-72,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :49
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(-80,0)"
            >
              <line
                y2="6"
                x2="0"
              />
              <text
                dy=".71em"
                style="text-anchor: middle;"
                y="9"
                x="0"
              >
                :50
              </text>
            </g>
            <path
              class="domain"
              d="M-80,6V0H0V6"
            />
          </g>
          <g
            class="axis y-axis"
          >
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-100)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                0.0
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-87.5)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                0.5
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-75)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                1.0
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-62.5)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                1.5
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-50)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                2.0
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-37.5)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                2.5
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-25)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                3.0
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,-12.5)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                3.5
              </text>
            </g>
            <g
              class="tick"
              style="opacity: 1;"
              transform="translate(0,0)"
            >
              <line
                x2="-6"
                y2="0"
              />
              <text
                dy=".32em"
                style="text-anchor: end;"
                x="-9"
                y="0"
              >
                4.0
              </text>
            </g>
            <path
              class="domain"
              d="M-6,-100H0V0H-6"
            />
          </g>
          <defs
            class="clip1"
          >
            <clippath
              id="clip1"
            >
              <rect
                width="-80"
                height="-100"
              />
            </clippath>
          </defs>
          <g
            clip-path="url(about:blank#clip1)"
            class="clipPath1"
          >
            <path
              class="line line1"
              stroke="#1f77b4"
              d="M0,-25L-80,0"
              style="stroke-opacity: 1;"
              stroke-dashoffset="0"
            />
          </g>
          <g>
            
          </g>
        </g>
        <g
          class="legend chart-legend"
          transform="translate(50,0)"
        >
          <g
            class="legend-group"
            transform="translate(0,0)"
          >
            <circle
              class="legend-circle"
              fill="#1f77b4"
              stroke="#1f77b4"
              fill-opacity="1"
              r="5"
            />
            <text
              class="legend-label"
              dx="10"
              dy="4"
            >
              reads
            </text>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;

exports[`storage resource time series should render empty data 1`] = `
<div>
  <div
    style="height: 500px; display: flex;"
  >
    <h3>
      No Data Available.
    </h3>
  </div>
</div>
`;

exports[`storage resource time series should render when fetching data 1`] = `
<div>
  <div
    style="height: 500px; display: flex;"
  >
    <h3>
      Fetching Data...
    </h3>
  </div>
</div>
`;
